### PR TITLE
Throw an exception which provides an info about vetoed property change

### LINKED ISSUE #3470: THe exception message is not informative: "For security reasons, the property must be defined as synchronized..."
When getting a server-side error at run time the error message below is missing the information on the element causing the issue:

```
java.lang.IllegalArgumentException: Feature 'com.vaadin.flow.internal.nodefeature.ElementPropertyMap' doesn't allow the client to update 'size'. For security reasons, the property must be defined as synchronized through the Element's API.
	at com.vaadin.flow.internal.nodefeature.ElementPropertyMap.deferredUpdateFromClient(ElementPropertyMap.java:96) ~[flow-server-1.0-SNAPSHOT.jar:na]
	at com.vaadin.flow.server.communication.rpc.MapSyncRpcHandler.handleNode(MapSyncRpcHandler.java:68) ~[flow-server-1.0-SNAPSHOT.jar:na]
	at com.vaadin.flow.server.communication.rpc.AbstractRpcInvocationHandler.handle(AbstractRpcInvocationHandler.java:60) ~[flow-server-1.0-SNAPSHOT.jar:na]
	at com.vaadin.flow.server.communication.ServerRpcHandler.handleInvocations(ServerRpcHandler.java:338) ~[flow-server-1.0-SNAPSHOT.jar:na]
	at com.vaadin.flow.server.communication.ServerRpcHandler.handleRpc(ServerRpcHandler.java:288) ~[flow-server-1.0-SNAPSHOT.jar:na]
	at com.vaadin.flow.server.communication.UidlRequestHandler.synchronizedHandleRequest(UidlRequestHandler.java:88) ~[flow-server-1.0-SNAPSHOT.jar:na]
	at com.vaadin.flow.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40) ~[flow-server-1.0-SNAPSHOT.jar:na]
	at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1516) ~[flow-server-1.0-SNAPSHOT.jar:na]
	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:342) [flow-server-1.0-SNAPSHOT.jar:na]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) [javax.servlet-api-3.1.0.jar:3.1.0]
```

Please add more details into the error message to make it possible to identify the component causing the issue.

### PR DESCRIPTION
**Pull request by denis-anisimov**
Fixes #3470

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/internal/nodefeature/PropertyChangeVetoException.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+package com.vaadin.flow.internal.nodefeature;
+
+/**
+ * The exception is thrown when a proposed change to a property is disallowed.
+ *
+ * @author Vaadin Ltd
+ *
+ */
+public class PropertyChangeVetoException extends Exception {
```
Comment (pleku) [inline flow-server/src/main/java/com/vaadin/flow/internal/nodefeature/PropertyChangeVetoException.java:24]: I would rename this to `PropertyChangeDeniedException` or `PropertyChangeDisallowedException`.
Veto is semantically right but it is still quite weird term to use and I think _denied_ would be best English and _ disallowed_ quite close too.

File: flow-server/src/main/java/com/vaadin/flow/internal/nodefeature/PropertyChangeVetoException.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.internal.nodefeature;
+
+/**
+ * The exception is thrown when a proposed change to a property is disallowed.
```
Comment (pleku) [inline flow-server/src/main/java/com/vaadin/flow/internal/nodefeature/PropertyChangeVetoException.java:19]:
```suggestion
 * The exception is thrown when a property change from client is disallowed on the server side.
```

Comment (denis-anisimov) [reply flow-server/src/main/java/com/vaadin/flow/internal/nodefeature/PropertyChangeVetoException.java:19]: Done

File: flow-server/src/main/java/com/vaadin/flow/internal/nodefeature/PropertyChangeVetoException.java (hunk 1 above)
Comment (denis-anisimov) [reply flow-server/src/main/java/com/vaadin/flow/internal/nodefeature/PropertyChangeVetoException.java:24]: Done .

JFYI:
https://docs.oracle.com/javase/7/docs/api/java/beans/PropertyVetoException.html

**Comment by pleku** [comment]
Tests not refactored with new exception name ?

**Comment by denis-anisimov** [comment]
> Tests not refactored with new exception name ?

Yes, Eclipse works with  Flow project worse every time....

**Comment by pleku** [comment]
As the bot is right and this adds new code -> fix is only to be cherry picked to `2.4` once we start building that. Otherwise it can stay just master.

**Pull request merged**